src/Nethermind/Nethermind.State/VerkleTransition/AccountTreeMigrator.cs

```diff
@@ -0,0 +1,229 @@
+    /// TODO: check if this is the best way to do this
+    /// </summary>
+    /// <param name="paddedHex"></param>
+    /// <returns></returns>
+    private static byte[] ConvertPaddedHexToBytes(string paddedHex)
+    {
+        // Remove the padding (leading zeros)
+        string unpadded = string.Concat(
+            paddedHex
+                .Where((c, i) => i % 2 == 1)
+                .Select(c => c.ToString())
+        );
+
+        // Convert the unpadded hex string to byte array
+        return Enumerable.Range(0, unpadded.Length / 2)
+            .Select(x => Convert.ToByte(unpadded.Substring(x * 2, 2), 16))
+            .ToArray();
+    }
+
+    private static byte[] ConstructFullHash(int branchIndex, byte[] nodeKey, List<byte[]> currentPath)
+    {
+        string branchPath = "";
+        if (!currentPath.IsNullOrEmpty())
+        {
+            branchPath = currentPath.ToArray().CombineBytes().ToHexString();
+        }
+        string branchIndexHex = branchIndex.ToString("x2");
+        string currentKey = nodeKey.ToHexString();
+        string addressHash = branchPath + branchIndexHex + currentKey;
+        string unpaddedAddressHash = ConvertPaddedHexToBytes(addressHash).ToHexString();
+
+        return Bytes.FromHexString(unpaddedAddressHash);
+    }
+
+    private static byte[] ConstructFullHash(byte[] nodeKey, List<byte[]> currentPath)
+    {
+        string branchPath = "";
+        if (!currentPath.IsNullOrEmpty())
+        {
+            branchPath = currentPath.ToArray().CombineBytes().ToHexString();
+        }
+        string currentKey = nodeKey.ToHexString();
+        string addressHash = branchPath + currentKey;
+        string unpaddedAddressHash = ConvertPaddedHexToBytes(addressHash).ToHexString();
+
+        return Bytes.FromHexString(unpaddedAddressHash);
+    }
+
+    public void VisitLeaf(TrieNode node, TrieVisitContext trieVisitContext, ReadOnlySpan<byte> value)
+    {
+        List<byte[]> currentPath = trieVisitContext.IsStorage ? _currentPathStorage : _currentPath;
+
+        while (currentPath.Count >= trieVisitContext.Level && trieVisitContext.Level > 0)
+        {
+            currentPath.RemoveAt(currentPath.Count - 1);
+        }
+
+        if (!trieVisitContext.IsStorage)
+        {
+            Account account = decoder.Decode(new RlpStream(node.Value.ToArray()));
+
+            // Reconstruct the full keccak hash
+            byte[] addressHash;
+            if (trieVisitContext.BranchChildIndex.HasValue)
+            {
+                addressHash = ConstructFullHash(trieVisitContext.BranchChildIndex.Value, node.Key, _currentPath);
+            }
+            else
+            {
+                addressHash = ConstructFullHash(node.Key, _currentPath);
+            }
+            byte[]? addressBytes = _preImageDb.Get(addressHash);
+            if (addressBytes is not null)
+            {
+                var address = new Address(addressBytes);
+                MigrateAccount(address, account);
```

Comment:
here just check if the account has code, then get the codeSize and update the account with the new codeSize